platform/mbed_cxxsupport.h

```diff
@@ -0,0 +1,1176 @@
+
+/* remove_pointer */
+template <typename T>
+struct remove_pointer : mbed::type_identity<T> { };
+
+template <typename T>
+struct remove_pointer<T *> : mbed::type_identity<T> { };
+
+template <typename T>
+struct remove_pointer<T * const>  : mbed::type_identity<T> { };
+
+template <typename T>
+struct remove_pointer<T * volatile> : mbed::type_identity<T> { };
+
+template <typename T>
+struct remove_pointer<T * const volatile> : mbed::type_identity<T> { };
+
+template <typename T>
+using remove_pointer_t = typename remove_pointer<T>::type;
+
+/* add_pointer */
+namespace impl {
+template <typename T>
+mbed::type_identity_t<remove_reference_t<T> *> add_pointer_helper(int);
+template <typename T>
+mbed::type_identity_t<T> add_pointer_helper(...);
+/* ARMC5 doesn't support class X : decltype(Y), so need intermediate class */
+template <typename T>
+using add_pointer_base = decltype(add_pointer_helper<T>(0));
+} // namespace impl
+
+template <typename T>
+struct add_pointer : impl::add_pointer_base<T> { };
+
+template <typename T>
+using add_pointer_t = typename add_pointer<T>::type;
+
+/* aligned_storage */
+namespace impl {
+constexpr size_t possible_alignment_requirement(size_t n)
+{
+    return n <= 1 ? 1 :
+           n <= 2 ? 2 :
+           n <= 4 ? 4 : 8;
+}
+}
+
+template <size_t Len, size_t Align = impl::possible_alignment_requirement(Len)>
+struct aligned_storage {
+    struct type {
+        __attribute__((aligned(Align))) unsigned char data[Len];
+    };
+};
+
+template <size_t Len, size_t Align = impl::possible_alignment_requirement(Len)>
+using aligned_storage_t = typename aligned_storage<Len, Align>::type;
+
+/* decay */
+namespace impl {
+template <typename T>
+struct decay : mbed::type_identity<
+                    conditional_t<is_array<T>::value,
+                        remove_extent_t<T>,
+                        conditional_t<is_function<T>::value,
+                            add_pointer_t<T>,
+                            remove_cv_t<T>
+                        >
+                    >
+                > { };
+} // namespace impl
+
+template <typename T>
+struct decay : impl::decay<remove_reference_t<T>> { };
+
+template <typename T>
+using decay_t = typename decay<T>::type;
+
+/* common_type not implemented */
+
+/* underlying_type */
+template <typename T>
+struct underlying_type : mbed::type_identity<__underlying_type(T)> { };
+
+template <typename T>
+using underlying_type_t = typename underlying_type<T>::type;
+
+/* result_of not implemented */
+
+/* addressof */
+template <typename T>
+T *addressof(T &arg) noexcept
+{
+    return reinterpret_cast<T *>(const_cast<char *>(&reinterpret_cast<const volatile char &>(arg)));
+}
+
+/* move */
+template <typename T>
+remove_reference_t<T> &&move(T &&a) noexcept
+{
+    return static_cast<remove_reference_t<T> &&>(a);
+}
+
+/* forward */
+template <typename T>
+T &&forward(remove_reference_t<T> &a) noexcept
```

Comment:
Same as above, it may be marked as constexpr. 
You should also provide the second overload: 

```c++
template <typename T>
 T &&forward(remove_reference_t<T> &&a) noexcept
```

Reply:
Hmm, lifted that from ARM C 5 manual, which only had the first overload. I wonder if there was any reason for that, or just an editorial error.